### PR TITLE
The Search for Modules: New filter to add module search terms

### PR DESCRIPTION
**Pull request by dereksmart**
Fixes: #1837 & #614

Adds a new filter for adding search terms to modules and increases current search range.  

Example usage: adding search terms for Stats, done in `module-info.php` like so: 
```
function jetpack_stats_search_terms( $terms ) {
	$terms = _x( 'statistics, tracking, analytics, views, traffic', 'search terms', 'jetpack' );
	return $terms;
}
add_filter( 'jetpack_search_terms_stats', 'jetpack_stats_search_terms' );
```

Search terms should be comma-separated and for best results, please translate with context using `_x()` to make it easier for our fellow GlotPressers down the road.

### CODE REVIEW

**Comment by samhotchkiss** [comment]
Looks good-- would love some feedback from @rase-

File: modules/module-info.php [hunk 1]
```diff
@@ -210,6 +212,11 @@ function stats_load_more_link( $description ) {
 }
 add_filter( 'jetpack_learn_more_button_stats', 'stats_load_more_link' );
 
+function jetpack_stats_search_terms() {
+	echo _x( 'statistics, tracking, analytics, views, traffic', 'search terms', 'jetpack' );
```
Comment (lezama) [inline modules/module-info.php]: this could use `_ex`

File: modules/module-info.php [hunk 2]
```diff
@@ -468,6 +475,11 @@ function jetpack_protect_more_link() {
 }
 add_action( 'jetpack_learn_more_button_protect', 'jetpack_protect_more_link' );
 
+function jetpack_protect_search_terms() {
+	echo _x( 'security, secure, protection, botnet, brute force', 'search terms', 'jetpack' );
```
Comment (lezama) [inline modules/module-info.php]: this could use `_ex`

File: class.jetpack-admin.php [hunk 3]
```diff
@@ -95,6 +95,10 @@ function get_modules() {
 				}
 				$module_array['long_description']  = ob_get_clean();
 
+				ob_start();
+				do_action( 'jetpack_search_terms_' . $module );
```
Comment (lezama) [inline class.jetpack-admin.php]: Why not use filters here?

Comment (dereksmart) [inline class.jetpack-admin.php]: No reason other than consistency with how we add info for other modules.
It might be useful though.  Should we make the long_description filterable also?

Comment (lezama) [inline class.jetpack-admin.php]: > Should we make the long_description filterable also?

yes, why not :)

Comment (dereksmart) [inline class.jetpack-admin.php]: May be best to do in a separate PR as to not take away the focus of Search Terms here

**Comment by rase-** [comment]
LGTM

**Comment by jeherve** [comment]
@dereksmart Could you add a docblock before the filter?

Thanks!

**Comment by dereksmart** [comment]
@jeherve Thanks for the reminder - done in f9d70c1

**Comment by jeherve** [comment]
Thanks!
